src/models/state/mempool/mempool_insertion_error.rs

```diff
@@ -0,0 +1,6 @@
+/// The reasons why inserting a transaction into the mempool may fail.
+#[derive(Debug, Clone, Copy, thiserror::Error)]
+pub(crate) enum MempoolInsertionError {
+    #[error("Too many inputs: got {got}, but only {allowed} are allowed.")]
+    TooManyInputs { got: usize, allowed: usize },
+}
```

Comment:
Too small and intricately tied to the mempool module to justify its own file.

Reply:
It might not be obvious from the code base but I rather like small files :)

Reply:
It's good to see more use of `thiserror` instead of `anyhow` in the library. I advocate for re-exporting all error types in a top-level `error.rs` to simplify discovery for downstream users.